### PR TITLE
Feature/new viewer

### PR DESCRIPTION
**Pull request by dmitry-sychugov**
Web: new viewer created.
Done: the ability to view images, audio and videos. Implemented functionality for the desktop: changing the volume and speed of the video, working with the context menu, flipping media. Ability to rotate, resize, and download.

Not done: the ability to rotate and increase the size of the image using the touch

### CODE REVIEW

File: packages/common/components/MediaViewer/sub-components/image-viewer.js [hunk 1]
```diff
@@ -249,10 +199,161 @@ class ImageViewer extends React.Component {
...
+          isDefaultMode={false}
+          directionY="top"
+          directionX="right"
+          fixedDirection={true}
+          withBackdrop={false}
+          manualY={(bottom || "63") + "px"}
+          manualX={(right || "-31") + "px"}
+        >
+          {model.map((item) => {
+            const onClick = (e) => {
```
Comment (gazizova-vlada) [inline packages/common/components/MediaViewer/sub-components/image-viewer.js:234]: Remove function onClick from render

File: packages/common/components/MediaViewer/sub-components/image-viewer.js [hunk 2]
```diff
@@ -249,10 +199,161 @@ class ImageViewer extends React.Component {
...
+                action={item.action}
+                onClick={onClick}
+              />
+            );
+          })}
+        </StyledDropDown>
+      );
+    };
+
+    var customToolbar = [
```
Comment (gazizova-vlada) [inline packages/common/components/MediaViewer/sub-components/image-viewer.js:253]: Change customToolbar variable to const

File: packages/components/viewer/index.js [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+  React.useEffect(() => {
+    document.addEventListener("mousemove", resetTimer);
+    return () => {
+      document.removeEventListener("mousemove", resetTimer);
+      clearTimeout(timer);
+      setPanelVisible(true);
+    };
+  }, []);
+
+  function resetTimer() {
```
Comment (gazizova-vlada) [inline packages/components/viewer/index.js:67]: Rewrite to arrow function

File: packages/components/viewer/index.js [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+    right: "0",
+    height: "53px",
+    background: "#333",
+    display: "flex",
+    justifyContent: "space-around",
+    alignItems: "center",
+  };
+
+  const mobileDetails = (
+    <div className="mobile-details" style={detailsStyle}>
```
Comment (gazizova-vlada) [inline packages/components/viewer/index.js:108]: Replace with styled components

File: packages/components/viewer/sub-components/icon.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  rotateRight: 6,
+  reset: 7,
+  close: 8,
+  scaleX: 9,
+  scaleY: 10,
+  download: 11,
+};
+
+export default function Icon(props) {
+  let prefixCls = "react-viewer-icon";
```
Comment (gazizova-vlada) [inline packages/components/viewer/sub-components/icon.js:18]: Replace with const

File: packages/components/viewer/sub-components/viewer-image.js [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+    };
+    props.onChangeImgState(
+      props.width,
+      props.height,
+      props.top + diffY,
+      props.left + diffX
+    );
+  }, [position]);
+
+  function handleResize(e) {
```
Comment (gazizova-vlada) [inline packages/components/viewer/sub-components/viewer-image.js:143]: Rewrite to arrow function

File: packages/components/viewer/sub-components/viewer-image.js [hunk 7]
```diff
@@ -0,0 +1,262 @@
...
+    props.onResize();
+  }
+
+  function handleMouseDown(e) {
```
Comment (gazizova-vlada) [inline packages/components/viewer/sub-components/viewer-image.js:147]: Rewrite to arrow function

File: packages/components/viewer/sub-components/viewer-image.js [hunk 8]
```diff
@@ -0,0 +1,262 @@
...
+  const handleMouseMove = (e) => {
+    if (isMouseDown.current) {
+      setPosition({
+        x: e.clientX,
+        y: e.clientY,
+      });
+    }
+  };
+
+  function handleResize(e) {
```
Comment (gazizova-vlada) [inline packages/components/viewer/sub-components/viewer-image.js:172]: Rewrite to arrow function

File: packages/components/viewer/sub-components/viewer-image.js [hunk 9]
```diff
@@ -0,0 +1,262 @@
...
+    props.onResize();
+  }
+
+  function handleMouseUp(e) {
```
Comment (gazizova-vlada) [inline packages/components/viewer/sub-components/viewer-image.js:176]: Rewrite to arrow function

File: packages/components/viewer/sub-components/viewer-image.js [hunk 10]
```diff
@@ -0,0 +1,262 @@
...
+
+  let styleIndex = {
+    zIndex: props.zIndex,
+  };
+
+  let imgNode = null;
+
+  if (props.imgSrc !== "") {
+    imgNode = (
+      <img
```
Comment (gazizova-vlada) [inline packages/components/viewer/sub-components/viewer-image.js:228]: Replace with styled components

File: packages/components/viewer/sub-components/viewer-image.js [hunk 11]
```diff
@@ -0,0 +1,262 @@
...
+        src={props.imgSrc}
+        style={imgStyle}
+        ref={imgRef}
+        onMouseDown={handleMouseDown}
+      />
+    );
+  }
+  if (props.loading) {
+    imgNode = (
+      <div
```
Comment (gazizova-vlada) [inline packages/components/viewer/sub-components/viewer-image.js:239]: Replace with styled components

File: packages/components/viewer/sub-components/viewer-toolbar.js [hunk 12]
```diff
@@ -0,0 +1,180 @@
...
+      content = config.render;
+    }
+
+    if (config.key === "percent") {
+      content = (
+        <div
+          className="iconContainer zoomPercent"
+          style={{ width: "auto", color: "#fff", userSelect: "none" }}
+        >
+          {`${props.percent}%`}
```
Comment (gazizova-vlada) [inline packages/components/viewer/sub-components/viewer-toolbar.js:100]: Need to destructure props in a component

File: packages/components/viewer/sub-components/viewer-image.js [hunk 13]
```diff
@@ -0,0 +1,262 @@
...
+  });
+  const [position, setPosition] = React.useState({
+    x: 0,
+    y: 0,
+  });
+
+  const handlers = useSwipeable({
+    onSwiping: (e) => {
+      const opacity =
+        props.scaleX !== 1 && props.scaleY !== 1
```
Comment (gazizova-vlada) [inline packages/components/viewer/sub-components/viewer-image.js:31]: Need to destructure props in a component

**Review by AlexeySafronov — APPROVED** [review]

**Pull request merged**